cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -122,6 +123,30 @@ func WriteKubeletConfigFiles(cfg *kubeadmapi.InitConfiguration, patchesDir strin
 	}
 
 	errs := []error{}
+
+	// We will first check if instance-config.yaml exists,
+	// and if it doesn't, then we get the containerRuntimeEndpoint from kubeadm-flags.env and write to instance-config.yaml
+	if features.Enabled(cfg.FeatureGates, features.NodeLocalCRISocket) {
+		_, err := os.Stat(filepath.Join(kubeletDir, kubeadmconstants.KubeletInstanceConfigurationFileName))
+		if os.IsNotExist(err) {
+			containerRuntimeEndpoint, err := kubeletphase.ReadKubeadmFlagsFromFile(filepath.Join(kubeletDir, kubeadmconstants.KubeletEnvFileName), "container-runtime-endpoint")
+			if err != nil {
+				errs = append(errs, errors.Wrap(err, "error reading kubeadm flags file"))
+			}
+
+			if err := kubeletphase.WriteInstanceConfigToDisk(containerRuntimeEndpoint, kubeletDir); err != nil {
+				errs = append(errs, errors.Wrap(err, "error writing instance kubelet configuration to disk"))
+			}
+
+			if dryRun { // Print what contents would be written
+				err = dryrunutil.PrintDryRunFile(kubeadmconstants.KubeletInstanceConfigurationFileName, kubeletDir, kubeadmconstants.KubeletRunDirectory, os.Stdout)
```

Comment:
best to print another like above that with `fmt.Fprintf(os.Stdout, kubeadmconstants.KubeletInstanceConfigurationFileName)`